nats.go

```diff
@@ -4341,6 +4375,7 @@ func (nc *Conn) removeSub(s *Subscription) {
 	}
 	// Mark as invalid
 	s.closed = true
+	s.changeSubStatus(SubscriptionClosed)
```

Comment:
Shouldn't we close the channel here to avoid leaks in use case when a lot of subscriptions are started and closed through the lifetime of the application?

Reply:
done